File: Year 3/Fall Semester/CS-315_Programming_Languages/Homework1/Codes/gultekin_omeroktay.py
print("--------------------------- 1 ---------------------------")
cs315_instructors = {
    "Section 1": "Altay Güvenir",
    "Section 2": "Aynur Dayanik",
    "Section 3": "Karani Kardas"
}

print("Initial CS 315 Instructors:", cs315_instructors)
print("--------------------------- 1 ---------------------------")
print()

print("--------------------------- 2 ---------------------------")
print("Instructor of Section 3 is {}".format(cs315_instructors["Section 3"]))
print("--------------------------- 2 ---------------------------")
print()

print("--------------------------- 3 ---------------------------")
cs315_instructors["Section 4"] = "Not Available"
print("CS 315 Instructors After Addition: {}".format(cs315_instructors))
print("--------------------------- 3 ---------------------------")
print()

print("--------------------------- 4 ---------------------------")
# print(del cs315_instructors["Section 4"])
del cs315_instructors["Section 4"]
print("CS 315 Instructors After Deletion:", cs315_instructors)
print("--------------------------- 4 ---------------------------")
print()

print("--------------------------- 5 ---------------------------")
cs315_instructors["Section 1"] = "Halil Altay Güvenir"
print("CS 315 Instructors After Modification:", cs315_instructors)
print("--------------------------- 5 ---------------------------")
print()

print("--------------------------- 6 ---------------------------")
print("Is Section 2 in CS 315 Sections:", "Section 2"
      in cs315_instructors.keys())
print("Is Section 1 in Cs 315 Sections:", "Section 1"
      in cs315_instructors)  # Also Valid
print("--------------------------- 6 ---------------------------")

print("--------------------------- 7 ---------------------------")
print("Is Karani Kardas one of instructors:", "Karani Kardas"
      in cs315_instructors.values())
print("--------------------------- 7 ---------------------------")

print()
print("--------------------------- 8 ---------------------------")
def foo(key, value):
    print(f"Key: {key}, Value: {value}")


print("Looping through associative array and printing key value pair:")
for item in cs315_instructors.items():
    foo(*item)

print()
print("Looping through associative array and printing key value pair (Second Way):")
for item in cs315_instructors.items():
    foo(item[0], item[1])

print()
print("Looping through associative array and printing key value pair (Third Way):")
for key, value in cs315_instructors.items():
    foo(key, value)

# Not work as expected, cs315_instructors returns only keys
# for key, value in cs315_instructors:
#    foo(key, value)
print("--------------------------- 8 ---------------------------")

File: Year 3/Fall Semester/CS-315_Programming_Languages/Homework1/Codes/gultekin_omeroktay.php
<?php
function foo($k, $v) {
  print "Key: $k Value: $v \n";
}

print "--------------------------- 1 ---------------------------" . "\n";
// as of PHP 5.4
$cs315Instructors = [
    "Section 1" => "Altay Güvenir",
    "Section 2" => "Aynur Dayanik",
    "Section 3" => "Karani Kardas"
];


// $search_array = array('first' => 1, 'second' => 4);
print_r($cs315Instructors);
print "--------------------------- 1 ---------------------------" . "\n";

print "--------------------------- 2 ---------------------------" . "\n";
print "Instructor of " . "Section 3 is ". $cs315Instructors["Section 3"] . "\n";
print "--------------------------- 2 ---------------------------" . "\n";

print "--------------------------- 3 ---------------------------" . "\n";
$cs315Instructors["Section 4"] = "Not Available";
print_r($cs315Instructors);
print "--------------------------- 3 ---------------------------" . "\n";

print "--------------------------- 4 ---------------------------" . "\n";
unset($cs315Instructors["Section 4"]);
print_r($cs315Instructors);
print "--------------------------- 4 ---------------------------" . "\n";

print "--------------------------- 5 ---------------------------" . "\n";
$cs315Instructors["Section 1"] = "Halil Altay Güvenir";
print_r($cs315Instructors);
print "--------------------------- 5 ---------------------------" . "\n";

print "--------------------------- 6 ---------------------------" . "\n";
print array_key_exists("Section", $cs315Instructors) . "\n";
print array_key_exists("Section 2", $cs315Instructors) . "\n";
print isset($cs315Instructors["Section 2"]) . "\n";
print "--------------------------- 6 ---------------------------" . "\n";

print "--------------------------- 7 ---------------------------" . "\n";
print (array_search("Karani Kardas", $cs315Instructors) == True) . "\n";
print "--------------------------- 7 ---------------------------" . "\n";

print "--------------------------- 8 ---------------------------" . "\n";
foreach($cs315Instructors as $k => $v) {
  foo($k, $v);
}
print "--------------------------- 8 ---------------------------" . "\n";
